heroic-core/src/main/java/com/spotify/heroic/metric/LocalMetricManager.java

```diff
@@ -713,32 +745,34 @@ public Histogram getRowDensityHistogram() {
         }
     }
 
-    private static class QuotaWatcher implements FetchQuotaWatcher, RetainQuotaWatcher {
+    private class QuotaWatcher implements FetchQuotaWatcher, RetainQuotaWatcher {
```

Comment:
What's the rationale for removing the `static`? Presumably this class is always referenced from within the enclosing class?

Reply:
I don't think so as we instantiating this class [here](https://github.com/spotify/heroic/blob/master/heroic-core/src/main/java/com/spotify/heroic/metric/LocalMetricManager.java#L371).

Reply:
@malish8632 this doesn't look right to me. In a multi-threaded env,  this will have some consequence on the number that is collected.  It is indeed always reference from this class.   You also need to consider  memory management implication.

Reply:
I've looked already into this with regards to memory - this object is instantiated rather than used as static.
Looking into original implementation the static method getLeft() was the only method marked as static but it was always used to compute on instance variables which to be honest doesn't make sense unless I'm missing something.
And to confirm that I inadvertently remove something important I've tested this code by running it for couple of days on production and can conclude that this change didn't affect much.